### ISSUE TITLE
Autogenerated matrices

### ISSUE THREAD

**Issue by MayorUshanka** [ballerburg9005/IQcaptcha#1]
Matrices usually have a multitude of any of these subpatterns:

- increasing rotation over (some) axis
- increasing scale in (some) direction over (some) axis
- ordering/permutation of some patterns, like a 1d sudoku

Sounds like this is automatable. Is this automation planned or has it not been considered?

**Comment by ballerburg9005** [comment]
Ok, so you you are asking if we can generate the Raven's matrices directly, instead of using fixed images?

There is another [iqcaptcha](https://github.com/kxgh/kx-iqcaptcha) where you could easily steal some code from. I would like to translate this code to PHP entirely, or at least have some sort of encoder in PHP on the server side and get rid of the node.js. 

This is one of the next steps to take if IQcaptcha use becomes more widespread.

**Comment by MayorUshanka** [comment]
Exactly. All of this overlay-noise image distortion crap just makes the captcha hard to read and tedious to solve. Whereas if you automatically generate the matrices, well there are no solution databanks, so good luck trying to bot it.

**Comment by MayorUshanka** [comment]
I can start working on a script that generates svg output

**Comment by ballerburg9005** [comment]
Ok, that would be nice thanks!

But it must be a server-side script and it can't have any "fancy" dependencies (such as node.js). 

I believe we will see the highest usage of IQcaptcha on Wordpress installations. And maybe even a little bit on .onion sites. That's why we should keep it PHP only.

I can solve [the other iqcaptcha](https://github.com/kxgh/kx-iqcaptcha) only half the time within the default 30 seconds or so. Maybe if it had one glyph less it would be easy enough, or maybe too easy as a default. I think we should take his image-generating code as an example.

**Comment by MayorUshanka** [comment]
Yeah, it doesn't seem to hard. Pick a couple of shapes, build a pattern, transform them, shuffle columns, remove one cell and put it in [answers], add some decoy answers.
